stdlib/public/libexec/swift-backtrace/TargetLinux.swift

```diff
@@ -0,0 +1,232 @@
+    name = Self.getProcessName(pid: pid)
+
+    let crashInfo: CrashInfo
+    do {
+      crashInfo = try reader.fetch(from: crashInfoAddr, as: CrashInfo.self)
+    } catch {
+      print("swift-backtrace: unable to fetch crash info.")
+      exit(1)
+    }
+
+    crashingThread = TargetThread.ThreadID(crashInfo.crashing_thread)
+    signal = crashInfo.signal
+    faultAddress = crashInfo.fault_address
+
+    images = Backtrace.captureImages(using: reader,
+                                     forProcess: Int(pid))
+
+    do {
+      try fetchThreads(threadListHead: Address(crashInfo.thread_list),
+                       limit: limit, top: top, cache: cache)
+    } catch {
+      print("swift-backtrace: failed to fetch thread information: \(error)")
+      exit(1)
+    }
+  }
+
+  /// Fetch information about all of the process's threads; the crash_info
+  /// structure contains a linked list of thread ucontexts, which may not
+  /// include every thread.  In particular, if a thread was stuck in an
+  /// uninterruptible wait, we won't have a ucontext for it.
+  func fetchThreads(
+    threadListHead: Address,
+    limit: Int?, top: Int, cache: Bool
+  ) throws {
+    var next = threadListHead
+
+    while next != 0 {
+      let t = try reader.fetch(from: next, as: thread.self)
+
+      next = t.next
+
+      guard let ucontext
+              = try? reader.fetch(from: t.uctx, as: ucontext_t.self) else {
+	// This can happen if a thread is in an uninterruptible wait
+        continue
+      }
+
+      let context = HostContext.fromHostMContext(ucontext.uc_mcontext)
+      let backtrace = try Backtrace.capture(from: context,
+                                            using: reader,
+                                            images: images,
+                                            limit: limit,
+                                            top: top)
+      guard let symbolicated
+              = backtrace.symbolicated(with: images,
+                                       sharedCacheInfo: nil,
+                                       useSymbolCache: cache) else {
+        print("unable to symbolicate backtrace for thread \(t.tid)")
+        exit(1)
+      }
+
+      threads.append(TargetThread(id: TargetThread.ThreadID(t.tid),
+                                  context: context,
+                                  name: getThreadName(tid: t.tid),
+                                  backtrace: symbolicated))
+    }
+
+    // Sort the threads by thread ID; the main thread always sorts
+    // lower than any other.
+    threads.sort {
+      return $0.id == pid || ($1.id != pid && $0.id < $1.id)
+    }
+
+    // Find the crashing thread index
+    if let ndx = threads.firstIndex(where: { $0.id == crashingThread }) {
+      crashingThreadNdx = ndx
+    } else {
+      print("unable to find the crashing thread")
+      exit(1)
+    }
+  }
+
+  public func redoBacktraces(limit: Int?, top: Int, cache: Bool) {
```

Comment:
IIUC, the type it's declared on is `internal`, do these `public` access modifiers have any effect here?

Reply:
No. Removed.